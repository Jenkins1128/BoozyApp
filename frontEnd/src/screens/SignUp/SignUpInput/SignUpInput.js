import React from 'react';
import { View, TextInput, Text, TouchableOpacity, StyleSheet } from 'react-native';

const SignUpInput = ({ state, dispatch, updateEmail, updatePassword, signupPressed }) => {
	return (
		<View style={styles.inputContainer}>
			<TextInput
				testID={'emailInput'}
				style={styles.emailInput}
				onChangeText={(email) => dispatch(updateEmail({ email: email }))}
				value={state.email}
				placeholder='Email'
				placeholderTextColor='#EB8873'
				underlineColorAndroid='transparent'
			></TextInput>
			<TextInput
				testID={'passwordInput'}
				style={styles.passwordInput}
				secureTextEntry={true}
				onChangeText={(password) => dispatch(updatePassword({ password: password }))}
				value={state.password}
				placeholder='Password'
				placeholderTextColor='#EB8873'
				underlineColorAndroid='transparent'
			></TextInput>
			<TouchableOpacity testID={'signupPressed'} onPress={signupPressed} style={styles.signUpButton}>
				<Text style={styles.signUpText}>Sign Up</Text>
			</TouchableOpacity>
		</View>
	);
};

const styles = StyleSheet.create({
	inputContainer: {
		flex: 3,
		width: '100%',
		paddingHorizontal: 40
	},
	emailInput: {
		borderRadius: 30,
		color: '#EB8873',
		padding: 30,
		backgroundColor: '#FFFFFF',
		borderTopWidth: 2,
		borderTopColor: '#ededed',
		marginBottom: 10
	},
	passwordInput: {
		borderRadius: 30,
		color: '#EB8873',
		padding: 30,
		backgroundColor: '#FFFFFF',
		marginBottom: 50
	},
	signUpButton: {
		width: '80%',
		height: 90,
		borderRadius: 70,
		alignItems: 'center',
		justifyContent: 'center',
		alignSelf: 'center',
		elevation: 8,
		color: '#fff',
		backgroundColor: '#EB8873'
	},
	signUpText: {
		color: 'white',
		fontSize: 24,
		fontWeight: 'bold'
	}
});

export default SignUpInput;
