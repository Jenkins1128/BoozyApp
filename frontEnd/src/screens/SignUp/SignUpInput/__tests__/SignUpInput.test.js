import React from 'react';
import renderer from 'react-test-renderer';
import SignUpInput from '../SignUpInput';

const state = {
	email: 'xxx@gmail.com'
};

describe('SignUpInput snapshot', () => {
	test('SignUpInput renders correctly', () => {
		const tree = renderer.create(<SignUpInput state={state} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
