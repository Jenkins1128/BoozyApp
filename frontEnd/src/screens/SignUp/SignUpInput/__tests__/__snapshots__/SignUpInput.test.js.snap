// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpInput snapshot SignUpInput renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 3,
      "paddingHorizontal": 40,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#EB8873"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 30,
        "borderTopColor": "#ededed",
        "borderTopWidth": 2,
        "color": "#EB8873",
        "marginBottom": 10,
        "padding": 30,
      }
    }
    underlineColorAndroid="transparent"
    value="xxx@gmail.com"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    placeholderTextColor="#EB8873"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 30,
        "color": "#EB8873",
        "marginBottom": 50,
        "padding": 30,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#EB8873",
        "borderRadius": 70,
        "color": "#fff",
        "elevation": 8,
        "height": 90,
        "justifyContent": "center",
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="signupButton"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Sign Up
    </Text>
  </View>
</View>
`;
